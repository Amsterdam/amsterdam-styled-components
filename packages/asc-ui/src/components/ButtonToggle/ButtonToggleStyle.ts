import styled from '@datapunt/asc-core'
import IconButtonStyle from '../IconButton/IconButtonStyle'
import { color } from '../../utils'

const BUTTON_SIZE = 50

type Props = {
  open: boolean
  size?: number
}
export default styled(IconButtonStyle)<Props>`
  height: ${({ size = BUTTON_SIZE }) => `${size}px`};
  width: ${({ size = BUTTON_SIZE }) => `${size}px`};
  background-color: ${({ open, theme }) =>
    color('tint', open ? 'level2' : 'level1')({ theme })};

  &:hover,
  &:focus {
    background-color: ${color('tint', 'level2')};
  }
`
