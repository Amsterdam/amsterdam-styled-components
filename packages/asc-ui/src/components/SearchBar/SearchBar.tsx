// import { Search } from '@amsterdam/asc-assets'
import {
  ChangeEvent,
  FormEvent,
  HTMLAttributes,
  // RefObject,
  useEffect,
  useState,
  forwardRef,
} from 'react'
import { deprecatedWarning } from '../../utils'
import Button from '../Button'
import Icon from '../Icon'
import { InputProps } from '../Input'
import InputContext from '../Input/InputMethodsContext'
import TextField, { TextFieldProps } from '../TextField/TextField'
import SearchBarStyle, { Props as SearchBarStyleProps } from './SearchBarStyle'

export interface SearchBarProps extends TextFieldProps, SearchBarStyleProps {
  label?: string
  inputProps?: InputProps
  /**
   * @deprecated, use onChange in combo with onClear instead
   * @param value
   */
  onWatchValue?: (value: string) => void
  /**
   * @deprecated, wrap this component inside a <form onSubmit={() => ...}> instead
   * @param value
   */
  onSubmit?: (e: FormEvent) => void
  autoFocus?: boolean
}

const SearchBar = forwardRef<
  HTMLInputElement,
  SearchBarProps & HTMLAttributes<HTMLInputElement>
>(
  (
    {
      children,
      placeholder,
      onSubmit,
      onBlur,
      onChange,
      onFocus,
      onWatchValue,
      value,
      label,
      hideAt,
      showAt,
      onClear,
      inputProps,
      autoFocus,
      ...otherProps
    },
    inputRef,
  ) => {
    // let inputRef: RefObject<HTMLInputElement> | null = null
    const [inputValue, setInputValue] = useState(value || '')

    console.log('search', inputRef)

    const handelOnClear = () => {
      setInputValue('')
      if (inputRef && inputRef.current) {
        inputRef.current.focus()
      }

      if (onClear) {
        onClear()
      }
    }

    const handleOnSubmit = (e: FormEvent) => {
      if (onSubmit) {
        deprecatedWarning(
          `onSubmit is about to be deprecated, wrap this component inside a <form onSubmit={...} /> instead`,
        )
        onSubmit(e)
      }
    }

    const handleOnChange = (e: ChangeEvent<HTMLInputElement>) => {
      setInputValue(e.currentTarget.value)

      if (onChange) {
        onChange(e)
      }
    }

    useEffect(() => {
      if (onWatchValue) {
        deprecatedWarning(
          'onWatchValue is about to be deprecated. Use onChange in combo with onClear instead',
        )
        // @ts-ignore
        onWatchValue(inputValue)
      }
    }, [inputValue, onWatchValue])

    useEffect(() => {
      if (typeof value !== 'undefined') {
        setInputValue(value)
      }
    }, [value, setInputValue])

    return (
      <SearchBarStyle {...otherProps} {...{ hideAt, showAt }}>
        <InputContext.Provider
          value={{
            onBlur,
            onFocus,
            onChange: handleOnChange,
            placeholder,
            ...inputProps,
            ...otherProps,
          }}
        >
          <TextField
            srOnly
            keepFocus
            blurOnEscape
            onClear={handelOnClear}
            aria-label={label}
            id={inputProps?.id}
            value={inputValue}
            autoFocus={autoFocus}
            {...{
              inputProps,
              ref: inputRef,
              label,
            }}
          />
        </InputContext.Provider>
        <Button
          aria-label={placeholder || 'Zoek'}
          title={placeholder || 'Zoek'}
          type="submit"
          size={36}
          onClick={handleOnSubmit}
          variant="secondary"
        >
          <Icon size={20}>
            <div />
          </Icon>
        </Button>
        {children}
      </SearchBarStyle>
    )
  },
)

SearchBar.defaultProps = {
  placeholder: 'Search...',
  value: '',
}

export default SearchBar
