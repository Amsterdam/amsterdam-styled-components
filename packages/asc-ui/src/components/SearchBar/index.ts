import SearchBarStyle from './SearchBarStyle'

export { default } from './SearchBar'

export const SearchBarStyles = {
  SearchBarStyle,
}
