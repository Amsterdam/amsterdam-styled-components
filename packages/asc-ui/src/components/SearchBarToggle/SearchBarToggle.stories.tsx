import React from 'react'
import { action } from '@storybook/addon-actions'
import { storiesOf } from '@storybook/react'
import { css } from '@datapunt/asc-core'
import SearchBarToggle from './SearchBarToggle'

const ControlledSearchBarToggle = () => {
  const [open, setOpen] = React.useState(false)
  return (
    <SearchBarToggle
      align="left"
      open={open}
      onOpen={setOpen}
      css={css`
        position: relative;
      `}
      onWatchValue={value => {
        action(`input value changed: ${value}`)
      }}
      onSubmit={() => {
        action('submit')
      }}
    />
  )
}

storiesOf('Composed/SearchBarToggle', module)
  .addDecorator(storyFn => (
    <div style={{ padding: '40px 10px' }}>{storyFn()}</div>
  ))
  .add('default', () => (
    <SearchBarToggle
      align="left"
      css={css`
        position: relative;
      `}
      onWatchValue={value => {
        action(`input value changed: ${value}`)
      }}
      onSubmit={() => {
        action('submit')
      }}
    />
  ))
  .add('controlled', () => <ControlledSearchBarToggle />)
