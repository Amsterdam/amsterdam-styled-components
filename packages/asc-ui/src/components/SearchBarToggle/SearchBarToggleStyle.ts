import styled, { css } from '@datapunt/asc-core'
import IconButtonStyle from '../IconButton/IconButtonStyle'
import { color } from '../../utils'
import SearchBarStyle from '../SearchBar/SearchBarStyle'

export type SearchBarMenuStyleProps = {
  open: boolean
  css?: string
}

const SearchBarMenuStyle = styled.div<SearchBarMenuStyleProps>`
  display: flex;
  align-items: flex-start;
  flex-direction: column;

  [aria-hidden='true'] {
    display: none;
  }

  & > ${SearchBarStyle} {
    position: absolute;
    top: 50px;
    right: 0;
    left: 0;
    padding: 15px;
    width: 100%;
    background-color: ${color('tint', 'level2')};
  }

  & > ${IconButtonStyle} {
    height: 50px;
    width: 50px;
    background-color: ${({ open, theme }) =>
      color('tint', open ? 'level2' : 'level1')({ theme })};
  }

  ${props =>
    props.css &&
    css`
      ${props.css}
    `}
`

export default SearchBarMenuStyle
