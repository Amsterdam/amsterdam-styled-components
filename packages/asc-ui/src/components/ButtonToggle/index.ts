import ButtonToggleStyle from './ButtonToggleStyle'

export { default } from './ButtonToggleStyle'

export { ButtonToggleStyle }
