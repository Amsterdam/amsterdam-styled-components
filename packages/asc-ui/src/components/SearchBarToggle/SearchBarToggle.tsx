import React from 'react'
import { Search } from '@datapunt/asc-assets'
import SearchBarToggleStyle, {
  SearchBarToggleStyleProps,
} from './SearchBarToggleStyle'
import SearchBar from '../SearchBar'
import { InputMethods, InputProps } from '../Input'
import Toggle, { Props as ToggleProps } from '../Toggle/Toggle'
import { SearchBarProps } from '../SearchBar/SearchBar'
import { CustomCssPropsInterface } from '../../utils'

interface SearchBarToggleProps
  extends SearchBarToggleStyleProps,
    InputMethods,
    CustomCssPropsInterface {
  placeholder?: string
  label?: string
  onSubmit?: Function
  onOpen?: Function
  open?: boolean
  inputProps?: InputProps
  searchBarProps?: SearchBarProps
  hasBackDrop?: boolean
}

const SearchBarToggle: React.FC<SearchBarToggleProps & ToggleProps> = ({
  children,
  hideAt,
  showAt,
  searchBarProps,
  inputProps,
  css,
  hasBackDrop,
  ...otherProps
}) => (
  <Toggle
    as={SearchBarToggleStyle}
    {...{
      hideAt,
      showAt,
      css,
      hasBackDrop,
    }}
    render={false}
    iconOpen={<Search />}
    {...otherProps}
  >
    <SearchBar
      inputProps={inputProps}
      focusOnRender
      {...searchBarProps}
      hasBackDrop={hasBackDrop}
    >
      {children}
    </SearchBar>
  </Toggle>
)

SearchBarToggle.defaultProps = {
  css: '',
  placeholder: 'Search...',
}

export default SearchBarToggle
