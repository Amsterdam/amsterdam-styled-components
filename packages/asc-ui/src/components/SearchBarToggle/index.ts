import SearchBarToggleStyle from './SearchBarToggleStyle'

export { default } from './SearchBarToggle'

export const SearchBarToggleStyles = {
  SearchBarToggleStyle,
}
