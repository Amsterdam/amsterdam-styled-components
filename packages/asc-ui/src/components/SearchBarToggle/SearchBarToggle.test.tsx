import * as React from 'react'
import { mount } from 'enzyme'
import SearchBar from '../SearchBar'
import SearchBarToggle from './SearchBarToggle'
import ButtonToggle from '../ButtonToggle'
import 'jest-styled-components'

jest.useFakeTimers()

describe('SearchBarToggle', () => {
  let component: any

  beforeEach(() => {
    const fn = jest.fn()
    component = mount(
      <SearchBarToggle onSubmit={fn} onChange={fn}>
        searchbar-content
      </SearchBarToggle>,
    )
  })

  it('should render', () => {
    expect(component).toMatchSnapshot()
  })

  it('it should show the searchbar when clicking on the button', () => {
    expect(component.find(SearchBar).exists()).toBe(false)
    component.find(ButtonToggle).simulate('click')
    expect(component.find(SearchBar).exists()).toBe(true)
  })
})
