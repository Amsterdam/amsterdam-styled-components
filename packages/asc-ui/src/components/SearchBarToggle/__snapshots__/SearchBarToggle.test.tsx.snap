// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarToggle should render 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 20px;
  height: 20px;
  box-sizing: content-box;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c3 > svg {
  width: inherit;
  height: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > .c4 {
  position: absolute;
  top: 50px;
  right: 0;
  left: 0;
  padding: 15px;
  border-bottom: 4px solid;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: normal;
  padding: 0 10px 0 10px;
  -webkit-transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 5px;
  width: 30px;
  height: 30px;
  height: 50px;
  width: 50px;
}

.c1:focus {
  z-index: 10;
  outline-style: solid;
  outline-offset: 0px;
  outline-width: 3px;
}

.c1:after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c1 .c2 {
  height: 20px;
  width: 20px;
}

<SearchBarMenu
  css=""
  onChange={[MockFunction]}
  onSubmit={[MockFunction]}
  placeholder="Search..."
>
  <styled.div
    css=""
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    open={false}
  >
    <StyledComponent
      css=""
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-brqgnP",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  align-items: flex-start;
  flex-direction: column;

  & > ",
              ".c4",
              " {
    position: absolute;
    top: 50px;
    right: 0;
    left: 0;
    padding: 15px;
    background-color: ",
              [Function],
              ";
    border-bottom: 4px solid;
  }

  ",
              [Function],
              "
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-brqgnP",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={
        Object {
          "current": .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 20px;
  height: 20px;
  box-sizing: content-box;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c3 > svg {
  width: inherit;
  height: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > .c4 {
  position: absolute;
  top: 50px;
  right: 0;
  left: 0;
  padding: 15px;
  border-bottom: 4px solid;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: normal;
  padding: 0 10px 0 10px;
  -webkit-transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 5px;
  width: 30px;
  height: 30px;
  height: 50px;
  width: 50px;
}

.c1:focus {
  z-index: 10;
  outline-style: solid;
  outline-offset: 0px;
  outline-width: 3px;
}

.c1:after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c1 .c2 {
  height: 20px;
  width: 20px;
}

<div
            class="c0"
          >
            <button
              aria-label="Search"
              class="c1"
              type="button"
            >
              <span
                class="c2 c3"
              >
                <svg
                  focusable="false"
                  viewBox="0 0 32 32"
                >
                  <path
                    d="M32 29.119l-9.825-9.825a12.217 12.217 0 1 0-2.88 2.881L29.118 32 32 29.119zm-19.771-8.746a8.145 8.145 0 1 1 0-16.289 8.145 8.145 0 0 1 0 16.29z"
                    fill="#000"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>,
        }
      }
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      open={false}
    >
      <div
        className="c0"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        open={false}
      >
        <Styled(Styled(styled.button))
          aria-label="Search"
          iconSize={20}
          onClick={[Function]}
          open={false}
          type="button"
        >
          <StyledComponent
            aria-label="Search"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-cMljjf",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  display: inline-flex;
  align-items: center;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: normal;
  padding: 0 10px 0 10px;
  ",
                    [Function],
                    "
  ",
                    "transition: color 0.1s ease-in-out, background-color 0.1s ease-in-out;",
                    "
  background: ",
                    [Function],
                    ";

  &:hover {
    background: ",
                    [Function],
                    ";
  }

  ",
                    "
  &:after {
    content: '';
    display: block;
    min-height: inherit;
    font-size: 0;
  }
",
                    "
",
                    "
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 5px;
  ",
                    [Function],
                    "

  ",
                    ".c2",
                    " {
    ",
                    [Function],
                    "
  }
",
                    "
  height: ",
                    [Function],
                    ";
  width: ",
                    [Function],
                    ";
  background-color: ",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "Styled(Styled(styled.button))",
                "foldedComponentIds": Array [
                  "sc-bdVaJa",
                  "sc-htpNat",
                ],
                "render": [Function],
                "styledComponentId": "sc-cMljjf",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            iconSize={20}
            onClick={[Function]}
            open={false}
            type="button"
          >
            <button
              aria-label="Search"
              className="c1"
              onClick={[Function]}
              open={false}
              type="button"
            >
              <Icon>
                <styled.span>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "c2",
                          "isStatic": false,
                          "lastClassName": "c3",
                          "rules": Array [
                            "
  display: ",
                            [Function],
                            ";
  ",
                            [Function],
                            "
  ",
                            [Function],
                            "
  ",
                            [Function],
                            ";
  box-sizing: content-box;
  ",
                            [Function],
                            ";

  & > svg {
    width: inherit;
    height: inherit;
    ",
                            [Function],
                            ";
  }
",
                          ],
                        },
                        "displayName": "styled.span",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "c2",
                        "target": "span",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <span
                      className="c2 c3"
                    >
                      <SvgSearch>
                        <svg
                          focusable={false}
                          viewBox="0 0 32 32"
                        >
                          <path
                            d="M32 29.119l-9.825-9.825a12.217 12.217 0 1 0-2.88 2.881L29.118 32 32 29.119zm-19.771-8.746a8.145 8.145 0 1 1 0-16.289 8.145 8.145 0 0 1 0 16.29z"
                            fill="#000"
                            fillRule="evenodd"
                          />
                        </svg>
                      </SvgSearch>
                    </span>
                  </StyledComponent>
                </styled.span>
              </Icon>
            </button>
          </StyledComponent>
        </Styled(Styled(styled.button))>
      </div>
    </StyledComponent>
  </styled.div>
</SearchBarMenu>
`;
